frontend/packages/ceph-storage-plugin/src/components/ocs-storage-class-form/ocs-storage-class-form.tsx

```diff
@@ -353,90 +384,72 @@ export const StorageClassEncryption: React.FC<ProvisionerProps> = ({
     namespace: CEPH_STORAGE_NAMESPACE,
     name: KMSConfigMapCSIName,
   };
-
   const [csiConfigMap, csiConfigMapLoaded, csiConfigMapLoadError] = useK8sWatchResource<
     ConfigMapKind
   >(csiCMWatchResource);
 
+  const setEncryptionId = React.useCallback(
+    (encryptionId: string) => {
+      setServiceName(encryptionId);
+      onParamChange(parameterKey, encryptionId, false);
+    },
+    [onParamChange, parameterKey],
+  );
+
+  /** When csiConfigMap is deleted from another tab "csiConfigMapLoadError" == true (404 Not Found),
+   * but, "csiConfigMap" still contains same old object that was present before the deletion of the configMap.
+   * Hence, dropdown was not updating dynamically. Used csiKmsDetails to handle that.
+   */
+  const [csiKmsDetails, setCsiKmsDetails] = React.useState<ConfigMapKind>(null);
   React.useEffect(() => {
-    if (isKmsSupported && !_.isEmpty(csiConfigMap)) {
-      const serviceNames = Object.keys(csiConfigMap?.data);
-      const kmsData = JSON.parse(csiConfigMap?.data[serviceNames[serviceNames.length - 1]]);
-      setCurrentKMS(kmsData);
-      const url = parseURL(kmsData.VAULT_ADDR);
-      const port = getPort(url);
-      const kmsObj: KMSConfig = {
-        name: {
-          value: kmsData.KMS_SERVICE_NAME,
-          valid: true,
-        },
-        address: {
-          value: `${url.protocol}//${url.hostname}`,
-          valid: true,
-        },
-        port: {
-          value: port,
-          valid: true,
-        },
-        backend: kmsData.VAULT_BACKEND_PATH,
-        caCert: state.kms.caCert ?? null,
-        caCertFile: state.kms.caCertFile,
-        tls: kmsData.VAULT_TLS_SERVER_NAME,
-        clientCert: state.kms.clientCert ?? null,
-        clientCertFile: state.kms.clientCertFile,
-        clientKey: state.kms.caCert ?? null,
-        clientKeyFile: state.kms.clientKeyFile,
-        providerNamespace: kmsData.VAULT_NAMESPACE,
-        hasHandled: true,
-      };
-      setEncryptionDispatch(SCActionType.SET_KMS_ENCRYPTION, '', dispatch, kmsObj);
-    } else {
-      setInProgress(false);
+    if (csiConfigMapLoaded && !csiConfigMapLoadError && csiConfigMap) {
+      setCsiKmsDetails(csiConfigMap);
+    }
+    // setting isExistingKms = false, when configMap not present (404 Not Found)
+    else if (csiConfigMapLoadError) {
+      setIsExistingKms(false);
+      setCsiKmsDetails(null);
+      setEncryptionId('');
     }
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [csiConfigMap, editKMS, isKmsSupported]);
+  }, [csiConfigMap, csiConfigMapLoaded, csiConfigMapLoadError, setIsExistingKms, setEncryptionId]);
 
   React.useEffect(() => {
-    if (isKmsSupported && (editKMS || !_.isEqual(state.kms, scInitialState.kms)))
+    if (isKmsSupported && !isExistingKms)
```

Comment:
```suggestion
    if (!isExistingKms)
```

Reply:
This whole React.useEffect is not required you can use state.hasHandled to decide to enable/disabled save button.